src/components/HeaderSearch/index.tsx

```diff
@@ -7,29 +7,33 @@ import Bind from 'lodash-decorators/bind'
 const styles = require('./index.less')
 
 interface HeaderSearchProps {
-  className: string
-  placeholder: string
-  open: boolean
-  onChange: Function
-  onPressEnter: Function
-  defaultOpen: boolean
+  placeholder?: string
+  dataSource?: string[]
+  defaultOpen?: boolean
+  open?: boolean
```

Comment:
are you sure this is optional? Should be required.

Reply:
Verified this again.
It is optional. Taken from ``interface IHeaderSearchProps`` which this components extends